src/Common/src/System/LocalAppContext.cs

```diff
@@ -35,29 +36,12 @@ private static bool GetCachedSwitchValueInternal(string switchName, ref int swit
             return isSwitchEnabled;
         }
 
-        private static bool DisableCaching 
-        { 
-            get 
+        private static bool DisableCaching =>
+            LazyInitializer.EnsureInitialized(ref s_disableCaching, ref s_isDisableCachingInitialized, ref s_syncObject, () =>
```

Comment:
This is for my learning. As with that at line 101, will this property access also allocate a delegate on each invocation? I wrote some test code for a static property and inspected it using ILSpy which generated following code,
```C#
private static HasDefaultCtor staticProperty
{
	get
	{
		Func<HasDefaultCtor> arg_2E_3;
		if ((arg_2E_3 = Program.<>c.<>9__6_0) == null)
		{
			arg_2E_3 = (Program.<>c.<>9__6_0 = new Func<HasDefaultCtor>(Program.<>c.<>9.<get_staticProperty>b__6_0));
		}
		return LazyInitializer.EnsureInitialized<HasDefaultCtor>(ref Program.s_defaultCtorObj, ref Program.s_hdcInited, ref Program.s_hdcLock, arg_2E_3);
	}
}
```
Looks like `arg_2E_3` will be allocated on each access for the static property as well? I am trying to understand how this is different from line 101 which also involves similar function call. Then would usage of LazyInitializer.EnsureInitialize with valueFactory always involve a delegate allocation, perhaps in those places (as this one) which was a plain function call previously? Please help me understand.